### PR TITLE
Refactoring grid column query

### PR DESCRIPTION
**Pull request by rprzedzik**

### CODE REVIEW

File: module/category/src/Domain/ValueObject/CategoryCode.php [hunk 1]
```diff
@@ -53,6 +53,7 @@ public function __toString(): string
      */
     public static function isValid(string $value): bool
     {
-        return strlen($value) <= 255;
+        return '' !== $value
+            && strlen($value) < 256;
```
Comment (WojFa) [inline module/category/src/Domain/ValueObject/CategoryCode.php]: what if `$value = '  ' `?

Comment (BastekBielawski) [reply module/category/src/Domain/ValueObject/CategoryCode.php]: You should use trim and then check it. Empty function is too risky.

File: module/category/src/Infrastructure/Grid/Builder/Query/CategorySystemAttributeDataSetQueryBuilder.php [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    /**
+     * @param QueryBuilder      $query
+     * @param string            $key
+     * @param AbstractAttribute $attribute
+     * @param Language          $language
+     *
+     */
+    public function addSelect(QueryBuilder $query, string $key, AbstractAttribute $attribute, Language $language): void
+    {
+        $query->addSelect(sprintf('(SELECT jsonb_agg(category_id) FROM product_category_product pcp WHERE pcp . product_id = p . id LIMIT 1) AS "esa_category:%s"', $language->getCode()));
```
Comment (BastekBielawski) [inline module/category/src/Infrastructure/Grid/Builder/Query/CategorySystemAttributeDataSetQueryBuilder.php]: Subselect in select is performace killer :)

File: module/category/tests/Application/Form/Type/CategoryTypeTest.php [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+namespace Ergonode\Category\Tests\Application\Form\Type;
+
+use Ergonode\Category\Application\Form\Type\CategoryType;
+use PHPUnit\Framework\TestCase;
+
+/**
+ */
+class CategoryTypeTest extends TestCase
+{
+
```
Comment (BastekBielawski) [inline module/category/tests/Application/Form/Type/CategoryTypeTest.php]: Empty test? :D

File: module/product/src/Infrastructure/Grid/Builder/DataSetQueryBuilder.php [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+    /**
+     * @param QueryBuilder      $query
+     * @param string            $key
+     * @param AbstractAttribute $attribute
+     * @param Language          $language
+     */
+    public function provide(QueryBuilder $query, string $key, AbstractAttribute $attribute, Language $language): void
+    {
+        foreach ($this->strategies as $strategy) {
+            if ($strategy->support($attribute)) {
```
Comment (BastekBielawski) [inline module/product/src/Infrastructure/Grid/Builder/DataSetQueryBuilder.php]: Should be `supports`, not `support`

File: module/product/src/Infrastructure/Grid/Builder/Query/AttributeDataSetQueryBuilderInterface.php [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+     * @return bool
+     */
+    public function support(AbstractAttribute $attribute): bool;
+
+    /**
+     * @param QueryBuilder      $query
+     * @param string            $key
+     * @param AbstractAttribute $attribute
+     * @param Language          $language
+     *
```
Comment (BastekBielawski) [inline module/product/src/Infrastructure/Grid/Builder/Query/AttributeDataSetQueryBuilderInterface.php]: Remove empty line

File: module/product/src/Infrastructure/Grid/Builder/Query/MultiSelectAttributeDataSetQueryBuilder.php [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+    {
+        return $attribute instanceof MultiSelectAttribute;
+    }
+
+    /**
+     * @param QueryBuilder      $query
+     * @param string            $key
+     * @param AbstractAttribute $attribute
+     * @param Language          $language
+     *
```
Comment (BastekBielawski) [inline module/product/src/Infrastructure/Grid/Builder/Query/MultiSelectAttributeDataSetQueryBuilder.php]: Remove empty line

File: module/product/src/Infrastructure/Grid/Builder/Query/MultiSelectAttributeDataSetQueryBuilder.php [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+use Ergonode\Attribute\Domain\Entity\AbstractAttribute;
+use Ergonode\Attribute\Domain\Entity\Attribute\MultiSelectAttribute;
+use Ergonode\Core\Domain\ValueObject\Language;
+
+/**
+ */
+class MultiSelectAttributeDataSetQueryBuilder implements AttributeDataSetQueryBuilderInterface
+{
+    /**
+     * @param AbstractAttribute $attribute
```
Comment (BastekBielawski) [inline module/product/src/Infrastructure/Grid/Builder/Query/MultiSelectAttributeDataSetQueryBuilder.php]: `{@inheritDoc}` will be better solution than copy paste

**Pull request merged**